pkg/controllers/applicationset_controller.go

```diff
@@ -167,12 +177,13 @@ func (r *ApplicationSetReconciler) deleteInCluster(ctx context.Context, applicat
 			err := r.Client.Delete(ctx, &app)
 			if err != nil {
 				appLog.WithError(err).Error("failed to delete Application")
-				continue
+				return err
```

Comment:
Instead of stoping the loop, we can try to do the best effort, and continue to delete other applications. We can use a boolean flag to indicate if there was an error

Reply:
agree

Reply:
Done. 

It seems to make sense here. We should be careful with the concept of continuing after an error though as there could be some implications, if we applied the same concept up a level and decided to proceed with cleanup of obsolete apps after a failure creating a new one, something unintended could happen.